sdk/core/src/main/java/com/klaviyo/coresdk/lifecycle/LifecycleMonitor.kt

```diff
@@ -0,0 +1,18 @@
+package com.klaviyo.coresdk.lifecycle
+
+import android.app.Activity
+
+typealias ActivityObserver = (activity: Activity) -> Unit
+
+/**
+ * Provides methods to react to changes in the application environment
+ */
+interface LifecycleMonitor {
+
+    /**
+     * Register an observer to be notified when all application activities stopped
+     *
+     * @param observer
+     */
+    fun whenStopped(observer: ActivityObserver)
```

Comment:
rename to `onAllActivitiesStopped` going off of android convention for observer names

Reply:
I was a little torn on that, I had started with `onStopped`. This class contains several `onXyz` methods that are invoked by the system after this object is with `ActivityLifecycleCallbacks`, whereas this method is for a _different_ service to register its closure as a callback to an event broadcast from _this_ object... Maybe I'm just overthinking it and the use case is clear from the arguments though.

Reply:
Nvm I decided I am indeed overthinking it.